soc/xtensa/intel_s1000/soc.c

```diff
@@ -186,13 +186,26 @@ u32_t soc_get_ref_clk_freq(void)
 
 static void soc_set_power_and_clock(void)
 {
-	volatile struct soc_dsp_shim_regs *regs =
-		(volatile struct soc_dsp_shim_regs *)
-		SOC_DSP_SHIM_REG_BASE;
-
-	regs->clkctl |= SOC_CLKCTL_REQ_FAST_CLK | SOC_CLKCTL_OCS_FAST_CLK;
-	regs->pwrctl |= SOC_PWRCTL_DISABLE_PWR_GATING_DSP1 |
+	volatile struct soc_dsp_shim_regs *dsp_shim_regs =
+		(volatile struct soc_dsp_shim_regs *)SOC_DSP_SHIM_REG_BASE;
+#if (CONFIG_AUDIO_INTEL_DMIC)
+	volatile struct soc_dmic_shim_regs *dmic_shim_regs =
+		(volatile struct soc_dmic_shim_regs *)SOC_DMIC_SHIM_REG_BASE;
+#endif
+
+	dsp_shim_regs->clkctl |= SOC_CLKCTL_REQ_FAST_CLK |
+		SOC_CLKCTL_OCS_FAST_CLK;
+	dsp_shim_regs->pwrctl |= SOC_PWRCTL_DISABLE_PWR_GATING_DSP1 |
 		SOC_PWRCTL_DISABLE_PWR_GATING_DSP0;
+
+#if (CONFIG_AUDIO_INTEL_DMIC)
+	/* enable power */
```

Comment:
You can move this section of code (202 to 207) to 193. That will ensure you use a single CONFIG_AUDIO_INTEL_DMIC which is more readable.

Reply:
Done.